Add initial docs on using Foreman MCP server

aneta-petrova opened the pull request:
#### What changes are you introducing?

Adding docs on using Foreman's MCP server.

#### Why are you introducing these changes? (Explanation, links to references, issues, etc.)

https://github.com/theforeman/foreman-mcp-server
https://community.theforeman.org/t/rfc-wrapper-plugin-for-ai-in-foreman/43284/4

#### Anything else to add? (Considerations, potential downsides, alternative solutions you have explored, etc.)

Considering how broad the use case is, I had a hard time finding a place in one of the existing guides. For example, Managing Hosts is not a good fit because the functionality covers more than just querying information about hosts (even though that's the only example use case we currently provide).

In the end, I decided to go with a new guide because 1. The content IMO doesn't fit any existing guide, for reasons stated above. 2. AI features gain a lot of traction and a separate guide could help with visibility. 3. This is just the initial MCP experience and the expectation is for more use cases to be developed, so it's likely we'll be adding more content in the future. (This proposal is to be discussed with Content Strategist.)

#### Contributor checklists

* [x] I am okay with my commits getting squashed when you merge this PR.
* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [ ] Foreman 3.15/Katello 4.17
* [ ] Foreman 3.14/Katello 4.16 (Satellite 6.17)
* [ ] Foreman 3.13/Katello 4.15 (EL9 only)
* [ ] Foreman 3.12/Katello 4.14 (Satellite 6.16; orcharhino 7.2 on EL9 only; orcharhino 7.3)
* [ ] Foreman 3.11/Katello 4.13 (orcharhino 6.11 on EL8 only; orcharhino 7.0 on EL8+EL9; orcharhino 7.1 with Leapp)
* [ ] Foreman 3.10/Katello 4.12
* [ ] Foreman 3.9/Katello 4.11 (Satellite 6.15; orcharhino 6.8/6.9/6.10)
* We do not accept PRs for Foreman older than 3.9.

#### Review checklists

Tech review (performed by an Engineer who did not author the PR; can be skipped if tech review is unnecessary):

* [ ] The PR documents a recommended, user-friendly path.
* [ ] The PR removes steps that have been made unnecessary or obsolete.
* [ ] Any steps introduced or updated in the PR have been tested to confirm that they lead to the documented end result.

Style review (by a Technical Writer who did not author the PR):

* [ ] The PR conforms with the team's style guidelines.
* [ ] The PR introduces documentation that describes a user story rather than a product feature.

aneta-petrova commented on guides/common/modules/con_overview-of-project-mcp-integration.adoc:
```diff
@@ -0,0 +1,9 @@
+[id="overview-of-project-mcp-integration"]
+= Overview of {Project} MCP integration
+
+{Project} provides a container image that you can use to run an MCP server locally on a {EL} server.
```
Dropped.

aneta-petrova commented on guides/common/modules/proc_configuring-the-project-mcp-server.adoc:
```diff
@@ -0,0 +1,39 @@
+Deploy the {Project} MCP server as a container and make it accessible to your MCP client.
+
+.Procedure
+. Build the MCP server container image.
+For example, to build the container from the current directory and assign the `foreman-mcp-server` tag to it:
++
+[options="nowrap", subs="+quotes,attributes"]
+----
+$ podman build --tag foreman-mcp-server .
+----
```
The image is not available in the registry yet, is that right?

aneta-petrova replied:
Updated, with a <TBD> in place of a step for satellite.

ekohl commented on guides/common/modules/proc_configuring-the-project-mcp-server.adoc:
```diff
@@ -0,0 +1,39 @@
++
+[options="nowrap", subs="+quotes,attributes"]
+----
+$ podman run --interactive --tty --publish 8080:8080 foreman-mcp-server \
+  --foreman-url https://_{foreman-example-com}_ \
+  --host localhost \
+  --port 8080 \
+  --transport streamable-http
```
I had more comments, but decided to file them as https://github.com/theforeman/foreman-mcp-server/issues/12 because they should be solved in the repo (and then used in documentation to simplify instructions).

aneta-petrova replied:
Going with the suggestion (even though I had to use a numerical IP address instead of localhost to get the container running).

We can update the step again later if https://github.com/theforeman/foreman-mcp-server/issues/12 gets resolved.

aneta-petrova commented on guides/common/modules/proc_querying-the-project-mcp-server.adoc:
```diff
@@ -0,0 +1,56 @@
+----
+<TBD> Show output. </TBD>
+----
+. Prompt: Narrow the list to systems in production environment.
```
In order to fix this up, I'll either need a complete test environment or a complete conversation output from someone who has that environment. I'm leaving this open and will look into it later.

aneta-petrova replied:
I used the examples from reporting.py. Thanks!

aneta-petrova commented on guides/common/modules/proc_configuring-the-project-mcp-server.adoc:
```diff
@@ -0,0 +1,39 @@
+  --transport streamable-http
+----
++
+The `--transport streamable-http` option ensures that the MCP server uses a streamable HTTP protocol for communication.
```
It might be better to refer to the MCP docs on transport modes.

aneta-petrova commented on guides/common/modules/proc_configuring-the-project-mcp-server.adoc:
```diff
@@ -0,0 +1,39 @@
+It enables MCP clients to query data from the server.
++
+The `--host localhost` option ensures that the MCP server binds to the localhost network interface.
+It makes the MCP server accessible only from the machine where the container is running.
```
Updated

aneta-petrova commented on guides/common/modules/proc_configuring-the-project-mcp-server.adoc:
```diff
@@ -0,0 +1,50 @@
+
+.Additional resources
+* `podman-run(1)` man page on your system
+* link:https://modelcontextprotocol.io/specification/2025-06-18/basic/transports[_Transports_ in the Model Context Protocol documentation]
```
This is really unfortunate: There is no way to link to a "latest" version of the Transports page. So I can either link to this version tied to a specific date, or the modelcontextprotocol.io introduction page.

aneta-petrova commented on guides/common/modules/proc_configuring-the-project-mcp-server.adoc:
```diff
@@ -0,0 +1,39 @@
+
+.Next steps
+* Configure your MCP client and connect it to the MCP server.
+<TBD> Is there anything we should tell users about this step? For example, anything on what information to provide to the MCP client to enable it to authenticate to Foreman? </TBD>
```
@adamruzicka And if using stdio as the transport mode, Foreman username and password must be set ([this line](https://github.com/theforeman/foreman-mcp-server/blob/03fb13fc8adab3db8943cff33917885cffd884cd/src/foreman_mcp_server/server.py#L37) shows the error message). What is the best way to provide these securely? [`--foreman-username` and `--foreman-password`](https://github.com/theforeman/foreman-mcp-server?tab=readme-ov-file#start-the-server-via-uv) doesn't look like something we should be recommending.

aneta-petrova replied:
> If using streamable-http as a transport, they will need to configure the client to send credentails as http headers.

In another comment, you recommended to document stdio as the 'more secure' transport method. I think that if we really think it's more secure, we shouldn't mention streamable-http at all. WDYT? (In the docs team, we have a guideline to document only one way of configuring things -- the one we recommend the most.)

pondrejk commented on guides/common/modules/proc_configuring-the-project-mcp-server.adoc:
```diff
@@ -0,0 +1,40 @@
+  --foreman-url https://_{foreman-example-com}_ \
+  --host 0.0.0.0 \
+  --port 8080 \
+  --transport stdio
```
This will work only if `FOREMAN_USERNAME` and `FOREMAN_PASSWORD` environment variables are set (`export FOREMAN_USERNAME=<usr>`, `export FOREMAN_PASSWORD=<pwd>`), which is probably the recommended way.  Otherwise the `--foreman-username` and `--foreman-password` options will be needed for for stdio transport.

pondrejk commented on guides/common/modules/con_overview-of-project-mcp-integration.adoc:
```diff
@@ -0,0 +1,9 @@
+{Project} provides a container image that you can use to run an MCP server locally.
+The {Project} MCP server can generate reports based on {Project} inventory.
+You can connect your MCP client to the {Project} MCP server to enable it to extract and process this inventory information.
+By querying the AI application through the MCP client, you can generate dynamic and comprehensive reports from your {Project} environment.
```
I'd emphasize that the MCP server has a read-only access to sat, so it can't make changes in there. Also I'd make note that the mcp server is meant to be "deployed" on a per user basis, meaning that everyone should deploy their own and use their own credentials

apinnick commented:
The Sat Admin guide might work for this documentation.

apinnick commented:
> I'd say this is aimed at users, not admins. We don't really have a user guide so I wonder what would be a good place for it.

@ekohl [Content journey](https://issues.redhat.com/browse/SAT-35535) says target persona is sys admin.

aneta-petrova commented:
That was exactly my line of thinking as well, as explained also in the PR's description. First, I'd like to get the content itself finalized (right now, we're still working on final definitions of the use cases and workflows so things will keep changing). I was going to initialize the conversation about placement after that -- once the content is closer to being ready. Let's wait a little more before making a decision :) Moving one or two assemblies should be pretty quick if we decide that a new guide is not the way to go.

aneta-petrova commented:
Hi @adamruzicka @ofedoren @pondrejk, I updated the draft based on the latest development. Can you please re-review and let me know if you have any comments or concerns?

Apart from this request, I also have two specific questions:

1. Do we have a path to the container downstream? For Satellite, the podman run and podman pull commands still contain only a placeholder.
2. Considering that answers to prompts might differ based on the LLM in use, do you think it even makes sense to provide example outputs in 2.3. Querying the Foreman MCP server?

aneta-petrova commented:
Dear writers, can I please get a style review? :)

sbernhard commented on guides/doc-Managing_Hosts/master.adoc:
```diff
@@ -60,6 +60,8 @@ ifdef::katello,orcharhino,satellite[]
 include::common/assembly_package-mode-and-image-mode-hosts.adoc[leveloffset=+1]
 endif::[]
 
+include::common/assembly_using-the-project-mcp-server.adoc[leveloffset=+1]
```
I think, we should disable this for orcharhino for now. We will have a deeper look into this area later on.

aneta-petrova replied:
Applied.

aneta-petrova commented:
In the latest version, after further discussion internally, I have removed the placeholders for example outputs. I have also removed the prerequisite to have multiple hosts registered because that was more about testing the user story rather than a documentation prerequisite.

adamruzicka commented on guides/common/modules/con_generating-reports-by-using-the-project-mcp-server.adoc:
```diff
@@ -0,0 +1,10 @@
+:_mod-docs-content-type: CONCEPT
+
+[id="generating-reports-by-using-the-project-mcp-server"]
+= Generating reports by using the {Project} MCP server
```
Afaik we won't really brand the mcp server itself, should we hardcode `{Project}` here? This would apply to multiple places here

adamruzicka commented on guides/common/modules/con_overview-of-project-mcp-integration.adoc:
```diff
@@ -0,0 +1,17 @@
+
+{Project} provides a container image that you can use to run an MCP server locally.
+{Project} MCP server is designed for advanced reporting and data analysis that leverages AI capabilities.
+You can use it to generate dynamic and comprehensive reports from your Satellite inventory.
```
```suggestion
You can use it to generate dynamic and comprehensive reports from your {Project} inventory.
```

adamruzicka commented on guides/common/modules/proc_configuring-the-mcp-client-to-connect-to-the-project-mcp-server.adoc:
```diff
@@ -0,0 +1,37 @@
+Perform these steps on the same system where you initiated the MCP server.
+
+.Prerequisites
+* Your MCP client must support HTTP headers.
```
```suggestion
* Your MCP client must support setting custom HTTP headers.
```

adamruzicka commented on guides/common/modules/proc_configuring-the-project-mcp-server.adoc:
```diff
@@ -0,0 +1,35 @@
++
+[options="nowrap", subs="+quotes,attributes"]
+----
+$ podman pull {mcp-server-image}
```
On Satellite, it may (or may not) be necessary to log into the registry before pulling the image. cc @ehelms

adamruzicka commented on guides/common/modules/proc_configuring-the-project-mcp-server.adoc:
```diff
@@ -0,0 +1,35 @@
++
+[options="nowrap", subs="+quotes,attributes"]
+----
+$ podman run --interactive --tty --publish localhost:8080:8080 \
```
```suggestion
$ podman run --interactive --tty --publish 127.0.0.1:8080:8080 \
```

otherwise I'm getting
```
Error: cannot parse "localhost" as an IP address
```

This also need to be changed in the paragraph below

adamruzicka commented on guides/common/modules/proc_configuring-the-project-mcp-server.adoc:
```diff
@@ -0,0 +1,35 @@
+----
+$ podman run --interactive --tty --publish localhost:8080:8080 \
+  {mcp-server-image} \
+  --foreman-url https://_{foreman-example-com}_ \
+  --host 0.0.0.0 \
+  --port 8080 \
+  --transport streamable-http
```
```suggestion
  --foreman-url https://_{foreman-example-com}_
```

The last three options match the defaults https://github.com/theforeman/foreman-mcp-server/pull/13/files

pondrejk commented on guides/common/modules/proc_querying-the-mcp-server-for-project.adoc:
```diff
@@ -0,0 +1,68 @@
+
+[NOTE]
+====
+While {Team} provides examples of prompts and responses, some LLMs might process these queries differently.
```
we have just examples of prompts

adamruzicka reviewed (commented):
One last nitpick, otherwise lgtm
